HWs/HW3/qsort.c

```diff
@@ -0,0 +1,191 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+#define arraySizeBound 1000
+#define sizeInsertionSort 10
+
+void printArray(int array[], int lengthArray)
+{
+    printf("[ ");
+    for (int i = 0; i < lengthArray; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("]");
+}
+
+void swap(int *firstNumber, int *secondNumber)
+{
+    int temp = *firstNumber;
+    *firstNumber = *secondNumber;
+    *secondNumber = temp;
+}
+
+int partition(int array[], int leftIndex, int rigthIndex)
+{
+    for (int current = leftIndex; current < rigthIndex; ++current)
+    {
+        if (array[rigthIndex] > array[current])
+        {
+            swap(&array[current], &array[leftIndex]);
+            ++leftIndex;
+        }
+    }
+
+    swap(&array[rigthIndex], &array[leftIndex]);
+
+    return leftIndex;
+}
+
+void insertsSort(int array[], int leftIndex, int rigthIndex)
+{
+    for (int i = leftIndex; i < rigthIndex; ++i)
+    {
+        int j = i + 1;
+
+        while (j > leftIndex && array[j - 1] > array[j])
+        {
+            swap(&array[j - 1], &array[j]);
+            --j;
+        }
+    }
+}
+
+void quickSort(int array[], int leftIndex, int rigthIndex)
+{
+    if (rigthIndex - leftIndex <= sizeInsertionSort - 2)
+    {
+        insertsSort(array, leftIndex, rigthIndex);
+    }
+    else
+    {
+        int reference = partition(array, leftIndex, rigthIndex);
```

Comment:
```suggestion
        int middleIndex = partition(array, leftIndex, rigthIndex);
```
Неговорящее имя переменной. На что ссылка?